src/v/raft/group_configuration.cc

```diff
@@ -277,6 +277,26 @@ unique_ids(const std::vector<vnode>& current, const std::vector<vnode>& old) {
     std::copy(unique_ids.begin(), unique_ids.end(), std::back_inserter(ret));
     return ret;
 }
+
+std::vector<vnode> build_replica_set(const raft::group_configuration& cfg) {
+    std::vector<vnode> ret;
+
+    const auto copy_unique = [&ret](const std::vector<vnode>& source) {
+        std::ranges::copy_if(
+          source, std::back_inserter(ret), [&ret](const vnode& vn) {
+              return std::ranges::find(ret, vn) == ret.end();
+          });
+    };
+
+    copy_unique(cfg.current_config().voters);
+    copy_unique(cfg.current_config().learners);
+    if (cfg.old_config()) {
+        copy_unique(cfg.old_config()->voters);
+        copy_unique(cfg.old_config()->learners);
+    }
+
+    return ret;
```

Comment:
nit: We only use this function when updating `cfg._all_replicas`. Make it a member function `void group_configuration::recalculate_all_replicas()` and include the assignment in there?